src/firefly/js/templates/lightcurve/LcPhaseFoldingPanel.jsx

```diff
@@ -408,6 +407,42 @@ function makeField1(hide) {
     return hide ? hidden : f1;
 }
 
+export function* listenerPanel() {
 
+    while (true) {
+        const action = yield take([
+            TABLE_HIGHLIGHT
+        ]);
 
+        switch (action.type) {
+            case TABLE_HIGHLIGHT:
+                handleTableHighlight(action);
+                break;
+        }
+    }
+}
+
+function handleTableHighlight(action) {
+    const {tbl_id} = action.payload;
+    const per = getPeriodFromTable(tbl_id);
+    if (per) {
+        dispatchValueChange({fieldKey: 'period', groupKey: grpkey, value: per});
+    }
+
+}
 //export default LcPhaseFoldingForm;
+
+/**
+ * gets the period from either peak or periodogram table, these tables doesn't necesary have a period column and same name
+ * @param {string} tbl_id
+ * @returns {string} period value
+ */
+function getPeriodFromTable(tbl_id) {
+    const tableModel = getTblById(tbl_id);
+    if (!tableModel || isNil(tableModel.highlightedRow)) return;
+    if ([PERIODOGRAM].includes(tbl_id)) {
```

Comment:
This is a bit confusing, why not tbl_id===PERIODOGRAM? Or you think we'll be adding more elements to this array in future?